mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td

```diff
@@ -260,74 +256,380 @@ def MIOpen_WorkitemIdOp:
   }];
+  let hasVerifier = 1;
+  let hasCanonicalizeMethod = 1;
 }
 
-// blockwise_store
-def MIOpen_BlockwiseStoreOp:
-    MIOpen_Op<"blockwise_store", [AttrSizedOperandSegments]>,
-    Arguments<(ins AnyMemRef:$dest,
-                  IndexArrayAttr:$bounds,
-                  ArgTransforms<1>:$transforms,
-                  Variadic<AnyTypeOf<[F32, F16, BF16, I8,
-                              VectorOfLengthAndType<[2, 4], [F32]>,
-                              VectorOfLengthAndType<[2, 4, 8], [F16]>,
-                              VectorOfLengthAndType<[2, 4, 8], [BF16]>,
-                              VectorOfLengthAndType<[4], [I8]>]>>:$data,
-                   Variadic<Index>:$destCoord)> {
-  let summary = "Blockwise GPU data store";
+// insert_slice
+def MIOpen_InsertSliceOp :
+    MIOpen_Op<"insert_slice", [NoSideEffect,
+      AllElementTypesMatch<["source", "dest"]>,
+      AllTypesMatch<["dest", "result"]>]>,
+    Arguments<(ins
+      AnyType:$source,
+      VectorOfRank<[1]>:$dest,
+      Index:$coord)>,
+    Results<(outs VectorOfRank<[1]>:$result)> {
+  let summary = "Insert a slice into a vector";
+
   let description = [{
-    The `miopen.blockwise_store` op moves data on GPU. Following movements are
-    allowed:
-    - Register (naive tensor) to LDS (naive tensor).
+    Inserts  `source` into contigious indices of `dest`, starting at `coord`.
+    If `source` is a scalar type, this is just `vector.insertelement`.
+    If the indices into `dest` go out of bounds, the result is undefined.
+  }];
+
+  let assemblyFormat = [{
+    attr-dict $source `->` $dest `[` $coord `]` `:` type($source) `->` type($dest)
+  }];
+  let hasVerifier = 1;
+  let hasCanonicalizeMethod = 1;
+}
+
+// yield
+def MIOpen_YieldOp :
+    MIOpen_Op<"yield", [NoSideEffect, Terminator, ReturnLike]> {
+  let summary = "yield for MIOpen loops";
+
+  let description = [{
+    Op that terminates the MIOpen looping constructs (currently transforming for).
+
+    If the loops produce any values, then this op's arguments should match the types
+    of those values. Otherwise, this op is implicit and can be omitted.
+
+    Note that, in a transforming_for, you don't need to yield back the lower coordinates,
+    as these are managed implicitly.
+  }];
+
+  // Yoinked from affine.yield
+  let arguments = (ins Variadic<AnyType>:$operands);
+
+  let builders = [
+    OpBuilder<(ins), [{ build($_builder, $_state, llvm::None); }]>
+  ];
+
+  let assemblyFormat = "attr-dict ($operands^ `:` type($operands))?";
+}
+
+// Transforming for
+def MIOpen_TransformingForOp :
+    MIOpen_Op<"transforming_for", [RecursiveSideEffects,
+      DeclareOpInterfaceMethods<LoopLikeOpInterface>,
+      AttrSizedOperandSegments,
+      SingleBlockImplicitTerminator<"::mlir::miopen::YieldOp">]>,
+    Arguments<(ins Variadic<Index>:$upperInits,
+      ArgTransformsAttr:$transforms,
+      Variadic<AnyType>:$iterInits,
+      IndexArrayAttr:$bounds,
+      I32ElementsAttr:$lowerStarts,
```

Comment:
It seems quite a few operations depends on the layout of this attribute. Could it be documented?